WIP: Geomap panel allows for selection of one query per layer

mmandrus opened the pull request:
**What this PR does / why we need it**:
Right now, the Geomap panel always uses the first query in the list, so users are not able to fully visualize all of their data. This fix allows users to select which query to use in each layer. It also includes some handling for when queries are renamed or deleted that will keep the UI up to date.

**Which issue(s) this PR fixes**:
Fixes #48274

**Special notes for your reviewer**:
Not happy yet with how the UI is kept up to date. Technically it works and is minimally hacky, but I feel like there has to be a better way that doesn't require re-initializing layers every time there's a datasource rename.

mmandrus committed:
add refId to frame comparison, since query names are turned into refIds

mmandrus committed:
create optional value override for select component

mmandrus committed:
implement value override in select component

mmandrus committed:
propogate query name changes up the panel editor heirarchy

mmandrus committed:
add new data query select with value override to the geomap options pane

mmandrus committed:
use selected data frame when updating a layer, rather than just the first

mmandrus committed:
Merge branch 'main' into mmandrus/geomap-data-layers

mmandrus committed:
move frame-replace logic into map panel so we can perform logic across all layers

mmandrus committed:
trigger a map update if a query is deleted

ryantxu commented on public/app/plugins/panel/geomap/layers/data/markersLayer.tsx:
```diff
@@ -112,43 +140,54 @@ export const markersLayer: MapLayerRegistryItem<MarkersConfig> = {
       legend: legend,
-      update: (data: PanelData) => {
-        if (!data.series?.length) {
+      update: (newData: PanelData, previousData?: PanelData, updateDataquery?: (newDataquery: string) => void) => {
```
rather than have the logic here... can it be up one layer?  That way each layer does not need to implement its own flavor?

ryantxu commented:
interesting!  most of this is getting the query editors to actually update their `refId`s 🤣

chri2547 committed:
updated alerts to alerting (#49446)

ryantxu commented:
I think the linked issue # is off, this does not show anything about feature toggles :)

FZambia committed:
Search: use prefix search for queries exceeding ngram filter max length (#49409)

ryantxu committed:
Search: scroll to the top after a query changes (#49447)

alexweav committed:
Alerting: Attach image URLs or upload files to Discord notifications. (#49439)

* Images in discord

* Drop duplicated field initialization

* Fix tests

* Use the proper context

adela-almasan committed:
Canvas: Quick positioning (#48992)

Co-authored-by: nmarrs <nathanielmarrs@gmail.com>

alexweav committed:
Alerting: Attach image URLs to Google Chat notifications. (#49445)

* Add a test card to see what styles look like

* Add text and image cards

* Address feedback

alexweav committed:
Alerting: Factor out shared code for working with configs in provisioning package (#49419)

* Factor out shared code for working with configs

* Refactor in notification policies and contact points

* Better file ordering

* Address feedback

nmarrs committed:
Canvas: Fix inconsistent element resizing UX (#49450)

FZambia committed:
search: close unused reader and writer on re-indexing (#49458)

vtorosyan committed:
Codeowners update: Map auth services to the Grafana AuthNZ team (#49421)

eleijonmarck committed:
Serviceaccounts: Do not display service accounts assigned to team (#48995)

* Add IsServiceAccount to query

* test for excluding service accounts in query

* align test

* test update

* added logging

* should fix tests

* moved test to bottom

* add back setup function

* Remove IsServiceAccount from query

Instead use false for is_service_account
and when the time comes to implement service accounts
for team. we deal with it

* missed teamtest

* formatting

Clarity-89 committed:
RolePicker: UI improvements  (#49397)

* RolePicker: Adjust input and display widths

* RolePicker: Use forwardRef

* RolePicker: Apply input styles to roles display

kaydelaney committed:
DashList: Handle long dashboard titles more gracefully (#49460)

kaydelaney committed:
VersionHistoryTable: Disable other checkboxes when two are selected (#49098)

ryantxu committed:
Search: remove special playground page and replace the standard results (#49442)

joanlopez committed:
Encryption: Enable envelope encryption by default (#49301)

* Encryption: Enable envelope encryption by default

* Stop relying on feature toggles from settings (deprecated)

* Database encryption docs (envelope encryption)

* Remove deprecated (and no longer used) FT

* Apply suggestions from code review

Co-authored-by: Tania <yalyna.ts@gmail.com>

Clarity-89 committed:
SaveDashboardForm: Ensure saveVariables is boolean (#49466)

joeblubaugh committed:
Alerting: Add entries to ngAlert changelog for screenshots. (#49457)

Notes the PRs that added the feature.

konrad147 committed:
Alerting: Use RadioButtonList component to select alert condition (#49422)

peterholmberg committed:
Alerting: Add notification policy flow chart (#49405)

* setup component add images

* add name to theme

* add root route info box

* remove unsused

* styling

yangkb09 committed:
Chore: Rename integration tests (#49438)

* Chore: Rename integration tests

* Remove one Integration

Co-authored-by: Ida Furjesova <ida.furjesova@grafana.com>

JoaoSilvaGrafana committed:
Chore: Converts SaveDashboardAsForm tests to RTL (#49436)

andresmgot committed:
AzureMonitor: Do not quote variables when a custom All variable option is used (#49428)

torkelo committed:
Variables: Fixes issue with data source variables not updating queries with variable (#49478)

gamab committed:
Devenv: Fix fake data gen to target mysql server (#49472)

gillesdemey committed:
Alerting: Make alertmanager datasource stable (#49485)

mildwonkey committed:
sqlstore: finish removing Find and SearchDashboards (#49347)

* chore: replace artisnal FakeDashboardService with generated mock

Maintaining a handcrafted FakeDashboardService is not sustainable now that we are in the process of moving the dashboard-related functions out of sqlstore.

* sqlstore: finish removing Find and SearchDashboards

Find and SearchDashboards were previously copied into the dashboard service. This commit completes that work, removing Find and SearchDashboards from the sqlstore and updating callers to use the dashboard service.

* dashboards: remove SearchDashboards from Store interface

SearchDashboards is a wrapper around FindDashboard that transforms the results, so it's been moved out of the Store entirely and the functionality moved into the Dashboard Service's search implementation.

The database tests depended heavily on the transformation, so I added testSearchDashboards, a copy of search dashboards, instead of (heavily) refactoring all the tests.

joshhunt committed:
Chore: Update theme json (#49480)

torkelo committed:
Search: Add primary background for global search container (#49473)

* Search: Add primarhy background for global search container

* Fix spacing for ActionRow

* Fixing ts issue

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 9.0.0-beta1 (#49489)

ashharrison90 committed:
Fix escape in Modal/DashboardSettings + add some unit tests (#49500)

alexanderzobnin committed:
DatePicker: able to set minimum date that can be selected (#49503)

wbrowne committed:
Alerting: Validate alert notification UID length (#45546)

* validate alert notification UID length

* simplify-ish

* remove unused field

* whoopsie

* delete newline

* remove check

* apply feedback

gillesdemey committed:
Alerting: Improve docs on alert state and health (#49410)

aocenas committed:
Prometheus: Migrate annotation editor to react (#48814)

* Modify the annotation support api

* Migrate annotation editor component

* Update public/app/features/annotations/standardAnnotationSupport.ts

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

* Move the escape hatches out of the public API

* Fix props transforms

* Break import cycle

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

kalleep committed:
AzureAD: Fallback to no groups if user does not have permission to query groups from azure  (#49506)

* If GroupMemver.Read.All permissions is not configured return no groups

* fix log

kaydelaney committed:
Search: Center FolderSection spinner (#49494)

yesoreyeram committed:
InfluxDB: Removing influxDB backend migration feature flag (#49531)

Removing the `influxdbBackendMigration` feature toggle default value.

kminehart committed:
upgrade grabpl (#49534)

achatterjee-grafana committed:
Docs: What's new 9.0 (#49453)

* initial commit

* Add to index list

* Improvements to intro section

* missing plural

* Minor fixes

* Some more changes.

* Updated image links

* Hide section "InfluxDB: Support for browser access mode removed

* Checkin more changes

* More changes.

* Fixed more image links

* Last batch of changes

* Fixed a few typos and ran prettier.

Co-authored-by: Torkel Ödegaard <torkel@grafana.com>

tombrk committed:
pkg/web: remove dependency injection (#49123)

* pkg/web: store http.Handler internally

* pkg/web: remove injection

Removes any injection code from pkg/web.

It already was no longer functional, as we already only injected into
`http.Handler`, meaning we only inject ctx.Req and ctx.Resp.

Any other types (*Context, *ReqContext) were already accessed using the
http.Request.Context.Value() method.

* *: remove type mappings

Removes any call to the previously removed TypeMapper, as those were
non-functional already.

* pkg/web: remove Context.Invoke

was no longer used outside of pkg/web and also no longer functional

gillesdemey committed:
Alerting: add "about alert rules" section (#49413)

* Alerting: add "about alert rules" section

* chore: adds descriptions and keywords

* Docs: update image locations

* chore: fix typo

toddtreece committed:
Prometheus: Streaming JSON parser performance improvements (#48792)

achatterjee-grafana committed:
Docs: Fixed typo in what's new in grafana 9 (#49549)

* Fixed typo

* Updated metadata

gillesdemey committed:
Docs: Add "about alerting" topic (#49523)

nmarrs committed:
Canvas: Fix resize breaking certain constraints (#49551)

wardbekker committed:
Prevent exception when panel has multiple queries (#49368)

This fixes the issues that when a panel has multiple queries this throws an exception if source is not part of the current frame

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

joanlopez committed:
Login: Fix AuthInfo update process (#49556)

* Login: Fix AuthInfo update process

* Fix GoDoc

* Add regression test for oauth info overwrite

Co-authored-by: jguer <joao.guerreiro@grafana.com>

idafurjes committed:
Add store split for Get Dashboard version method (#49138)

* Add store split for Get Dashboard version method

* Implement dashboard version service

* Fix api tests

* Remove GetDashboarVersion from sqlstore

* Add fakes for Get dashboard version

* Fix sqlstore test

* Add Get Dashboard store test

* Add dashver service test

* Remove useless comments

gabor committed:
devenv: better loki environment (#49383)

* devenv: rename loki to loki-promtail

* devenv: new loki devenv entry with fake data generation

torkelo committed:
AngularSupport: Update description for angular_support_enabled config option (#49569)

* AngularSupport: Update description for angular_support_enabled config option

* Update angular deprecration plan doc

* Update

* Update article

* Updated

Clarity-89 committed:
TeamList: Convert test to RTL (#49567)

ifrost committed:
Chore: Convert a test from enzyme to testing library (#49492)

ifrost committed:
Query History: Track query history migration failures (#49560)

dprokop committed:
Old Graph migration: Preserve null values settings (#49493)

* Graph migration: Preserve null values settings

* Review

ashharrison90 committed:
don't close drawer when clicking a select within it (#49583)

joshhunt committed:
Cue: Update Text, Stat, and Annolist panel cue schemas (#49479)

* user essentials mob! :trident:

* user essentials mob! :trident:

* user essentials mob! :trident:

lastFile:public/app/plugins/panel/stat/types.ts

* user essentials mob! :trident:

* user essentials mob! :trident:

* Use generate defaults, and fix a few remaining issues

* Add annolist to dashboard dist plugins

* minor fixes

Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>
Co-authored-by: Joao Silva <joao.silva@grafana.com>
Co-authored-by: Alexandra Vargas <alexa1866@gmail.com>
Co-authored-by: sam boyer <sdboyer@grafana.com>

IevaVasiljeva committed:
Access control: fix a typo for folder actions (#49582)

* fix a typo for folder actions

* update dashboard action too

kalleep committed:
AccessControl: Grant data source reader to all users when running oss (#49514)

* grant data source reader to all users when running oss or enterprise
without license

* fix asserts in alerting tests

* add oss licensing service for test setup

* fix tests to pass in enterprise

* lint

* fix tests

* set setting.IsEnterprise flag for tests

Co-authored-by: Yuriy Tseretyan <yuriy.tseretyan@grafana.com>

kalleep committed:
AccessControl: Pass current org id to UsersTable (#49584)

* Pass current org id to UsersTable

* Mock contextSrv and update snapshot

marefr committed:
Chore: Add no-changelog label for backports (#49605)

aangelisc committed:
CloudMonitor: Correctly encode default project response (#49510)

* Correctly encode default project response

* Make getGCEDefaultProject a method of Service and add test

* Handle error appropriately

* Update test and function definition

ashharrison90 committed:
Cue: Use BarGauge, DashList and Gauge panel cue schemas (#49580)

* user essentials mob! :trident:

* Wip

* user essentials mob! :trident:

lastFile:public/app/plugins/panel/bargauge/module.tsx

* user essentials mob! :trident:

* update betterer results file

Co-authored-by: kay delaney <kay@grafana.com>
Co-authored-by: Levente Balogh <balogh.levente.hu@gmail.com>
Co-authored-by: Joao Silva <joao.silva@grafana.com>

chri2547 committed:
Docs: get started updates (#49351)

* adjusts topic titles and file names

* adjusts weight

* corrects typo

* publishes sql server topic

* adjusts topic titles, heading size

peterholmberg committed:
Alerting: Add legacy indicator to navbar (#49511)

* update id and add translation key

* remove logs

itsmylife committed:
Add cypress type support (#49315)

DanCech committed:
fix broken alias due to #48635 (#49623)

ashharrison90 committed:
Apply .page-toolbar class correctly (#49603)

ashharrison90 committed:
convert Typeahead test to RTL (#49509)

yuri-tceretian committed:
show notification tab only to editor (#49624)

ryantxu committed:
Search: add icon for row and singlestat panels (#49620)

toddtreece committed:
Prometheus: Enable prometheusStreamingJSONParser by default (#49475)

connorlindsey committed:
Explore: Add ability to include tags in trace to metrics queries (#49433)

* Add tags input

* Tracing: add ability to include tags in trace to metrics queries

nmarrs committed:
Canvas: Improve changing element options UX (#49555)

kalleep committed:
AccessControl: Replace IsEnterprise checks with license checks (#49572)

axelavargas committed:
SearchV2 - Fix starred dashboards for new organizations error (#49645)

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

achatterjee-grafana committed:
Updated document with Michelle Tan's comments (#49648)

gguillotte-grafana committed:
Docs: Identify which Grafana editions are relevant to each Enterprise doc (#49207)

* Add section to Ent docs index re: Cloud features
* Add and update notes identifying Enterprise and Cloud features
* Address feedback

hairyhenderson committed:
Util: Improve performance of strings.SplitString (#49115)

Replaces the regexp with calls to strings.ReplaceAll and strings.Fields
for simplicity and improved performance.

Signed-off-by: Dave Henderson <dave.henderson@grafana.com>

ryantxu committed:
Timeseries: fix outside range stale state (#49633)

Co-authored-by: Todd Treece <todd.treece@grafana.com>

achatterjee-grafana committed:
Docs: Created separate section for migration under alerting (#49616)

* Lots of changes, including new topic, also fixed all alerting relrefs

* Push stashed change.

* Updated content to reflect that Grafana alerting is enabled during upgrade.

* Fix typo

* Update docs/sources/alerting/migrating-alerts/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Checkin changes.

* Updates from Chris's review.

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

achatterjee-grafana committed:
Docs: Move alert rule section to alerting fundamentals section (#49657)

* initial commit

* Added links to alert rules, and fixes one broken alerting relref.

toddtreece committed:
Prometheus: Fix sort issue in wide frames (#49660)

joeblubaugh committed:
Alerting: Add stored screenshot utilities to the channels package. (#49470)

Adds three functions:
`withStoredImages` iterates over a list of models.Alerts, extracting a stored image's data from storage, if available, and executing a user-provided function.
`withStoredImage` does this for an image attached to a specific alert.
`openImage` finds and opens an image file on disk.

Moves `store.Image` to `models.Image`
Simplifies `channels.ImageStore` interface and updates notifiers that use it to use the simpler methods.
Updates all pkg/alert/notifier/channels to use withStoredImage routines.

gabor committed:
loki: added two new functions (#49617)

eleijonmarck committed:
refactoring: saml (#48114)

Co-authored-by: Vardan Torosyan <vardants@gmail.com>

gabor committed:
loki: better unpack handling (#49074)

konrad147 committed:
Narrow the alert condition picker (#49570)

jackw committed:
Build: Enable long term caching for frontend assets (#47625)

* build(webpack): move CopyUniconsPlugin into own file

* chore(webpack): delete unused blobUrl and compile loaders

* build(webpack): prefer contenthash over fullhash for longer caching

* build(webpack): set optimization.moduleIds named only in dev

* build(webpack): introduce HTMLWebpackCSSChunks so templates can access theme css by name

* feat: inject css files with contenthash in html templates

* revert(error-template): remove ContentDeliveryURL from CSS href

* refactor(index-template): update grafanaBootData.themePaths

* chore(webpack): add typescript annotations for CopyUniconsPlugin

JoaoSilvaGrafana committed:
TimeRange: Fixes issue when zooming out on a timerange with timespan 0 (#49622)

ifrost committed:
Chore: Re-enable skipped test (#49484)

* Re-enable skipped test

alexanderzobnin committed:
Fix wrap_handler() panic during OAuth login (#49671)

JoaoSilvaGrafana committed:
Variables: Fixes issue with data source variables not being correctly used in panel queries (#49323)

leventebalogh committed:
fix: only "swallow" the json() parsing error if the response is empty (#47493)

ashharrison90 committed:
Notification history: Enable by default (#49502)

* remove persistNotifications feature flag

* remove unused imports

ashharrison90 committed:
ValueMapping: Add support for regex replacement over multiple lines (#49607)

Jguer committed:
Remove migrations from behind configs (#49415)

gelicia committed:
Explore: Use Dashboard permissions in Explore To Dashboard (#49539)

* Look for dashboard permissions in e2d button

* Get tests to pass

* Add permissions tests

papagian committed:
Alerting: Fix swagger specification (#49273)

* Alerting: fix specification

* Update merged swagger specification

vtorosyan committed:
Quick Ugly Fix for Oauth not to make web_hack.go panic (#49653)

ivanahuckova committed:
Preferences: Fix updating of preferences for Navbar and Query History (#49677)

* Preferences: Fix PUT for Navbar and Query History

* Update linting

* Fix linting

* Use models defined in preference package in preferences

* Remove white space

* Revert "Use models defined in preference package in preferences"

This reverts commit 6f7aa9f8d35a7aad505fa3bad7e2ddeaddb4c527.

Elfo404 committed:
Explore: Reset Graph overrides if underlying series changes (#49680)

gabor committed:
Loki code editor: do not run query on blur in explore mode (#49241)

* loki code editor: do not run query on blur in explore mode

* loki: code editor: better change tracking

* fixed comment

Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

kaydelaney committed:
Chore: Remove usage of deprecated getColorForTheme function (#49519)

toddtreece committed:
Docs: Fix duplicate text in What’s new in Grafana v9.0 (#49699)

toddtreece committed:
Docs: Document Prometheus NaN breaking change (#49688)

aangelisc committed:
AzureMonitor: Ensure original query properties are included unchanged (#49642)

* Ensure original query properties are included unchanged

* Remove refId from pseudo datasources

* Include testing for interpolateVariablesInQueries

- Add util function to create template variables
- Update mock query with missing props
- Additional tests on each ds for template variables

* Correct typo and add explicit check for datasource

ashharrison90 committed:
convert bargauge test to RTL (#49701)

kaydelaney committed:
Chore: Remove deprecated TextDisplayOptions export (#49705)

ifrost committed:
Query History: Add feature tracking for query history (#49685)

* Add feature tracking for query history

* Fix unit tests

* Update event names

jackw committed:
Toolkit: Bump dependencies (#47826)

* chore(typescript): remove sanitize_url.d.ts in favour of npm package types

* chore(toolkit): bump all webpack related deps inline with grafana core

* refactor(toolkit): prefer webpack types and use from imports for bundling

* feat(toolkit): bundle plugins using webpack 5 and babel

* chore(toolkit): but all eslint deps inline with grafana core

* refactor(toolkit): rewrite linting step to use new eslint node api

* chore(toolkit): bump jest dependencies inline with grafana core

* refactor(toolkit): update jest config for jest 27

* fix(toolkit): resolve toolkit if using yarn berry

* docs(toolkit): update instructions for developing with yarn berry

* chore(toolkit): remove yarnlink code as won't work with yarn berry

* chore(toolkit): bump remaining dependencies

* chore(toolkit): remove unused core.start task

* feat(toolkit): use browserlist when building plugins

* chore(toolkit): add browserslist dependency

* refactor(toolkit): resolve style loaders, update postcss options for webpack5

* chore(toolkit): put back grafana/data and grafana/ui

* docs(toolkit): improve instructions for developing toolkit locally

* chore(toolkit): clean up webpack debug and warnings

* chore(input-datasource): remove pnpwebpack plugin and update browserslist to solve failing build

* chore(renovatebot): remove toolkit package.json from ignorePaths

* revert(renovate): put back toolkit package.json in ignorePaths

* feat(toolkit): introduce babel plugins

* refactor(toolkit): remove runtime automatic from preset-react for earlier versions of react

* refactor(toolkit): add missing fallbacks to webpack config

* fix(toolkit): remove spaces from copy webpack glob so files are copied

* refactor(toolkit): fix up babel typescript support and copy browserlist on build

* chore(yarn): refresh lock file

* revert(toolkit): remove browsers list so plugins compile to ES5

* revert(toolkit): remove copying .browserslistrc

jdbaldry committed:
Convert TOML front matter to YAML (#49690)

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

ashharrison90 committed:
convert PartialHighlighter test to RTL (#49708)

kaydelaney committed:
Chore: Convert legacy `Input` test to RTL (#49713)

j6s committed:
Alerting: Do not include button in googlechat notification if URL invalid (#47317)

* Alerting: Do not include button in googlechat notification if URL invalid

* Apply suggestions from code review

Co-authored-by: Alexander Weaver <weaver.alex.d@gmail.com>

* Alerting: Add test case for invalid external URL in googlechat notifier

Co-authored-by: Alexander Weaver <weaver.alex.d@gmail.com>

nmarrs committed:
Canvas: Fix element placement UX (#49665)

alexweav committed:
Alerting: Endpoints for provisioning mute timings (#49635)

* Add validator for mute timing and make it provisionable

* Add tests to ensure prometheus validators are running and errors are propagated

* Internal API for manipulating mute timings

* Define and generate API layer

* Wire up generated code

* Implement API handlers

* Tests for golang layer

* Fix reference bug

* Fix linter and auth tests

* Resolve semantic errors and regenerate

* Remove pointless comment

* Extract out provisioning path param keys, simplify

* Expected number of paths

iwysiu committed:
CloudWatch: Use scopedVars in expressions (#49178)

* CloudWatch: Use scopedVars in expressions

* fix spec test

Commit:
devnev: Update several dashes to schemaVersion 36 (#48976)

This is just to make a more robust set of test inputs test suite for
schema validation.

kaydelaney committed:
Chore: Remove deprecated `surface` prop from IconButton (#49715)

* Chore: Remove deprecated `surface` prop from IconButton

* Update component docs

ryantxu committed:
SearchV2: improve searcher API, and include a fallback SQL based implementation (#49535)

Commit:
schema: Generate Go and Typescript from Thema coremodels  (#49193)

* Add go code generator for coremodels

* Just generate the entire coremodel for now

Maybe we'll need more flexibility as more coremodels are added, but for
now this is fine.

* Add note on type comment about stability, grodkit

* Remove local replace directive for thema

* Generate typescript from coremodel

* Update pkg/coremodel/dashboard/addenda.go

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

* Update cuetsy to new release

* Update thema to latest

* Fix enum generation for FieldColorModeId

* Put main generated object at the end of the file

* Tweaks to generated Go output

* Retweak back to var

* Add generated coremodel test

* Remove local replace statement again

* Add Make target and call into cuetsy cmd from gen

* Rename and comment linsrc for readability

* Move key codegen bits into reusable package

* Move body of cuetsifier into codegen pkg

Also genericize the diffing output into reusable WriteDiffer.

* Refactor coremodel generator to use WriteDiffer

* Add gen-cue step to CI

* Whip all the codegen automation into shape

* Add simplistic coremodel canonicality controls

* Remove erroneously committed test

* Bump thema version

* Remove dead code

* Improve wording of non-canonicality comment

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

joeblubaugh committed:
Alerting: Add tests for Slack Notifier Image Upload (#49669)

Tests that file bytes are correctly read from storage when an image is
available without a URL. Tests that not found and success cases are both
handled.

ivanahuckova committed:
Loki: Fix unwrap parsing (#49732)

chri2547 committed:
Docs: combines all SAML topics into one topic (#49443)

* combines all SAML topics

* resolves merge conflicts

* makes prettier

chri2547 committed:
Docs/fixes relrefs (#49754)

* fixes http-api link

* removes old saml docs

* fixes broken links to enterprise docs landing page

achatterjee-grafana committed:
Docs: Refactor contact points section of alerting docs (#49758)

* Initial commit

* weight adjustments

* Checkin changes.

achatterjee-grafana committed:
Docs: Update create alerting rules topics (#49764)

* Updated topic on Grafana managed rule.

* Updated loki and mimir rules and recording rules topics

* Checkin changes.

* Fixed typo

torkelo committed:
PanelChrome: Fixes issue with padding not being applied as gridUnits (#49751)

eleijonmarck committed:
Docs: integration tests for using sqllite (#49455)

* Update developer-guide.md

* prettier formatting

wbrowne committed:
tidy up (#49720)

grobinson-grafana committed:
Alerting: Add GetImages to ImageStore (#49717)

* Alerting: Add GetImages to ImageStore

* Use assert.ElementsMatch instead of sort.Sort

ivanahuckova committed:
AutoSizeInput: Move to @grafana/ui (#48811)

* AutoSizeInput: Move to @grafana/ui

* Update packages/grafana-ui/src/components/AutoSizeInput/AutoSizeInput.mdx

Co-authored-by: JitaC <70489351+achatterjee-grafana@users.noreply.github.com>

* Update packages/grafana-ui/src/components/AutoSizeInput/AutoSizeInput.mdx

Co-authored-by: JitaC <70489351+achatterjee-grafana@users.noreply.github.com>

* Update packages/grafana-ui/src/components/AutoSizeInput/AutoSizeInput.mdx

Co-authored-by: JitaC <70489351+achatterjee-grafana@users.noreply.github.com>

* Fix linter error

* Update packages/grafana-ui/src/components/AutoSizeInput/AutoSizeInput.mdx

Co-authored-by: JitaC <70489351+achatterjee-grafana@users.noreply.github.com>

* Update packages/grafana-ui/src/components/AutoSizeInput/AutoSizeInput.mdx

Co-authored-by: JitaC <70489351+achatterjee-grafana@users.noreply.github.com>

* Move AutoSizeInput to Input folder

* Use iconOptions in storybook

Co-authored-by: JitaC <70489351+achatterjee-grafana@users.noreply.github.com>

kaydelaney committed:
Search: Fix overflow issue with folder view (#49723)

ashharrison90 committed:
add event tracking to navigation (#49800)

ashharrison90 committed:
cap mobile nav height (#49785)

jdbaldry committed:
Fix typo in curl command (#46729)

* Fix typo in curl command

Co-authored-by: eleijonmarck <eric.leijonmarck@gmail.com>
Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

* Fix typo in another curl command

Co-authored-by: eleijonmarck <eric.leijonmarck@gmail.com>
Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

Co-authored-by: eleijonmarck <eric.leijonmarck@gmail.com>

ashharrison90 committed:
Chore: Upgrade to jest 28 (#49679)

* bump packages to jest 28

* changes needed for jest 28

* map react-colorful as well

* use customResolver and fix last test

* don't need react-colorful installed if we're using a custom resolver

* return correct thing in mock

* remove watchPathIgnorePatterns since we don't have node_modules anymore

andresmgot committed:
AzureMonitor: Document maximum returned records for Resource Graph (#49801)

dependabot[bot] committed:
Bump actions/upload-artifact from 2 to 3 (#47541)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 2 to 3.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Clarity-89 committed:
RolePicker: Add extra offset for menu position (#49823)

toddtreece committed:
Prometheus: Disable prometheusStreamingJSONParser by default (#49825)

asimpson committed:
AzureMonitor: migrate Logs+Arg QueryEditor to @grafana/experimental (#48978)

* AzureMonitor: use @grafana/experimental for LogsQueryEditor

ref #48711

* AzureMonitor: migrate ArgQueryEditor to @grafana/experimental.

Change CodeEditor to 100% instead of 1000px to match LogsQueryEditor.

ref #48713

zoltanbedi committed:
Fix: Timeseries migration regex override (#49629)

torkelo committed:
devenv: Fix data source uid for default gdev-testdata (#49820)

zoltanbedi committed:
Chore: Convert FolderPicker test to rtl (#49838)

kostrse committed:
Prometheus: Predefined scopes for Azure authentication (#49557)

* Predefined scopes for Azure Prometheus

* Allow override of audience

grobinson-grafana committed:
Alerting: Enable Unified Alerting for open source and enterprise (#49834)

This commit enables Unified Alerting for open source and enterprise unless disabled in configuration.

kalleep committed:
AccessControl: Cleanup access control interface (#49783)

* removed unused function

* Rename interface

JacobsonMT committed:
Alerting: Add templated subject config to email notifier (#49742)

* Add subject templating to email notifier

* Fix linting

joshhunt committed:
Dashboard: Fix panel menu styling issues (#49806)

* Dashboard: Fix panel menu styling issues

* remove !importants I added

kaydelaney committed:
ClipboardButton: Simplify callbacks (#49847)

axelavargas committed:
SearchV2: Fix search input width on mobile (#49850)

* SearchV2: Fix search input on mobile

kyamau committed:
Alerting: Chore: Fix event name in ngalert/README.md (#49829)

Fix event name from *evalContext to *evaluation as #45144.

leeoniya committed:
HeatmapNG: ensure non-zero y bucket size (#49454)

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

JoaoSilvaGrafana committed:
PanelEditor: Fixes issue with Table view and multi data frames (#49854)

torkelo committed:
Angular: Remove deprecated angular modal support and libs (#49781)

joey-grafana committed:
Elasticsearch: Migrate annotation editor to react (#49529)

* Migrate annotation editor to react

* Gio patch

* Update types

* Optional and conform to older design

Clarity-89 committed:
TeamMemberRow: Convert tests to RTL (#49875)

* TeamMemberRow: Remove legacy select

* TeamMemberRow: Update attributes

* TeamMemberRow: Refactor tests to RTL

kaydelaney committed:
UI/Button: Remove deprecated "link" variant (#49843)

dsotirakis committed:
Dependencies: Update etcd module (#49873)

* Update etcd

* Update go.sum

* Replace etcd due to broken module

toddtreece committed:
Prometheus: Predefined scopes for Azure authentication (#49861)

idafurjes committed:
Remove preferences from models (#49879)

renovate[bot] committed:
Update dependency react-inlinesvg to v3 (#49244)

* Update dependency react-inlinesvg to v3

* Remove unused property

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Zoltán Bedi <zoltan.bedi@gmail.com>

aocenas committed:
Prometheus: Cleanup annotation editor (#49615)

* Remove unused code

* Remove test

* Remove Builder mode and simplify the code

* Fix step mapping

* Fix import

* change placeholder

idafurjes committed:
Split Delete expired dashboard version store (#49610)

* Split Delete expired dashboard version store

* Add method to fakes

* Fix lint

* Fix lint 2

* Use split store method in cleanup

* Add tests

* Remove DeleteExpiredVersions from sqlstore

* Fix lint

* Fix integration tests

gillesdemey committed:
Alerting: fix layout with long words / numbers (#49882)

kalleep committed:
add migrator to drop folder create actions that was set fromt he folder (#49878)

kaydelaney committed:
UI/Card: Remove deprecated props (#49885)

ashharrison90 committed:
add feature highlighting in the expanded menu (#49892)

kalleep committed:
AuthProxy: Remove deprecated ldap_sync_ttl setting (#49902)

* Remove deprecated ldap_sync_ttl

konrad147 committed:
Alerting: Add fgac support to alert list panel (#47934)

* Add fgac support to alert list panel

* Fix typo

* test: fix test suite for UnifiedAlertStatesWorker

* Update worker permission check

* Update alert worker tests

Co-authored-by: gillesdemey <gilles.de.mey@gmail.com>

zoltanbedi committed:
Chore: convert NewWindowIcon test to rtl (#49830)

L-M-K-B committed:
grafana/ui: Move panel-container styles to component (#49566)

* Chore: transfer styling to Explore.tsx and convert to Emotion

* feat: create a component and export it

* Chore: replace by new component

* Chore: replace by new component

* Feat: create a story

* Chore: clean up

* Chore: clean up

joeblubaugh committed:
Alerting: Add Go error message to warning log for screenshots. (#49870)

Makes debugging problems with alert screenshotting easier.

zoltanbedi committed:
Chore: remove jest-coverage-badges dep from toolkit (#49883)

Clarity-89 committed:
TeamMembers: Refactor test to RTL (#49922)

kaydelaney committed:
ColorPicker: Remove deprecated onColorChange prop (#49923)

jdbaldry committed:
Use ref links for external content (#49786)

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

marefr committed:
Datasource: Fix allowed cookies to be forwarded as header to backend datasources (#49541)

Co-authored-by: Will Browne <wbrowne@users.noreply.github.com>

kaydelaney committed:
InlineLabel: Remove deprecated props (#49929)

JoaoSilvaGrafana committed:
Chore: Converts DashboardRow tests to RTL (#49722)

marefr committed:
Plugins: Support headers field for check health (#49930)

polibb committed:
Convert Enzyme UsersTable test to RTL (#49836)

* Convert Enzyme UsersTable test to RTL

* Fixes on UsersTable test; no additions

* remove snapshot of UsersTable test

wbrowne committed:
regenerate swagger spec (#49938)

marefr committed:
Datasource: Remove deprecated max_idle_connections_per_host setting (#49948)

mmandrus committed:
Merge branch 'mmandrus/geomap-data-layers' of github.com:grafana/grafana into mmandrus/geomap-data-layers

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=49444) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=49444) before we can accept your contribution.<br/>**52** out of **53** committers have signed the CLA.<br/><br/>:white_check_mark: alexanderzobnin<br/>:white_check_mark: leventebalogh<br/>:white_check_mark: JoaoSilvaGrafana<br/>:white_check_mark: ashharrison90<br/>:white_check_mark: Jguer<br/>:white_check_mark: papagian<br/>:white_check_mark: vtorosyan<br/>:white_check_mark: Elfo404<br/>:white_check_mark: gabor<br/>:white_check_mark: ivanahuckova<br/>:white_check_mark: gelicia<br/>:white_check_mark: aangelisc<br/>:white_check_mark: jdbaldry<br/>:white_check_mark: ifrost<br/>:white_check_mark: j6s<br/>:white_check_mark: jackw<br/>:white_check_mark: toddtreece<br/>:white_check_mark: kaydelaney<br/>:white_check_mark: sdboyer<br/>:white_check_mark: nmarrs<br/>:white_check_mark: joeblubaugh<br/>:white_check_mark: iwysiu<br/>:white_check_mark: chri2547<br/>:white_check_mark: alexweav<br/>:white_check_mark: achatterjee-grafana<br/>:white_check_mark: ryantxu<br/>:white_check_mark: wbrowne<br/>:white_check_mark: zoltanbedi<br/>:white_check_mark: dependabot[bot]<br/>:white_check_mark: Clarity-89<br/>:white_check_mark: eleijonmarck<br/>:white_check_mark: andresmgot<br/>:white_check_mark: grobinson-grafana<br/>:white_check_mark: asimpson<br/>:white_check_mark: torkelo<br/>:white_check_mark: JacobsonMT<br/>:white_check_mark: axelavargas<br/>:white_check_mark: leeoniya<br/>:white_check_mark: dsotirakis<br/>:white_check_mark: joey-grafana<br/>:white_check_mark: joshhunt<br/>:white_check_mark: idafurjes<br/>:white_check_mark: kyamau<br/>:white_check_mark: kostrse<br/>:white_check_mark: kalleep<br/>:white_check_mark: mmandrus<br/>:white_check_mark: L-M-K-B<br/>:white_check_mark: polibb<br/>:white_check_mark: gillesdemey<br/>:white_check_mark: marefr<br/>:white_check_mark: aocenas<br/>:white_check_mark: konrad147<br/>:x: renovate[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=49444) it.</sub>

mmandrus commented:
Wow... I did something horrific apparently. Closing this and starting over
